tests/unit/Framework/MockObject/Matcher/ConsecutiveParametersTest.php

```diff
@@ -65,4 +65,38 @@ public function testIntegrationExpectingException(): void
 
         $mock->foo('invalid');
     }
+
+    /**
+     * Test for the issue https://github.com/sebastianbergmann/phpunit/issues/3590
+     */
+    public function testMutableObjectsChangeSuccess()
+    {
+        /** @var \DateTime|MockObject $b */
+        $b = $this->getMockBuilder(\DateTime::class)
+            ->setMethods(['diff'])
+            ->getMock();
+
+        $validationValues = [];
+
+        $b->expects($this->exactly(2))
+        ->method('diff')
+            ->withConsecutive([
+                $this->callback(function (\DateTime $it) use (&$validationValues) {
+                    $validationValues[0] = $it->format('Y');
+                    return $it->format('Y') === '2019';
+                })],
+                [$this->callback(function (\DateTime $it) use (&$validationValues)  {
+                    $validationValues[1] = $it->format('Y');
+                    return $it->format('Y') === '1970';
+                })]
+            );
+
+        $arg = \DateTime::createFromFormat('Y-m-d', '2019-01-01');
```

Comment:
If possible, please use `\DateTimeImmutable` instead of `\DateTime`. In this case this means using two separate objects (instead of using `setDate()`).

Reply:
I am not sure it makes sense to do that here, as soon as this unit-test replicates the scenario described within the issue #3590 description, where the expected call does change the object state, therefore must be performed on the same object.